### PR TITLE
Add round trip XAML testing

### PR DESCRIPTION
**Pull request by dmetzgar**
This PR goes toward issue #19. Part of the original framework tests was to take a given activity and populate its properties with randomized values then serialize to XAML, deserialize, and compare the objects before and after. There is some special casing, such as for arguments. Not all cases are handled yet and some activities use dynamic implementations that currently are not serializable with the test code.

This code will also be helpful for testing that the System.Xaml library used on non-Windows platforms works correctly.

### CODE REVIEW

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp3.0</TargetFrameworks>
+
+    <IsPackable>false</IsPackable>
+
+    <RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
```
Comment (lbargaoanu) [inline src/Test/TestCases.Xaml/TestCases.Xaml.csproj]: Just remove these two I guess.

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (lbargaoanu) [inline src/Test/TestCases.Xaml/TestCases.Xaml.csproj:1]: We need to do smth about the XamlTest projects. Perhaps rename the existing one to TestCases.Workflows?

File: src/CoreWf/Expressions/AndAlso.cs [hunk 3]
```diff
@@ -10,7 +10,9 @@ namespace System.Activities.Expressions
 using System.Activities.DynamicUpdate;
 
     //[SuppressMessage(FxCop.Category.Naming, FxCop.Rule.IdentifiersShouldNotMatchKeywords, Justification = "Optimizing for XAML naming. VB imperative users will [] qualify (e.g. New [AndAlso])")]
+#pragma warning disable CA1716 // Identifiers should not match keywords
```
Comment (lbargaoanu) [inline src/CoreWf/Expressions/AndAlso.cs]: Wouldn't [this](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute?view=netframework-4.8) work? Or any other less painful syntax :)

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp3.0</TargetFrameworks>
```
Comment (lbargaoanu) [inline src/Test/TestCases.Xaml/TestCases.Xaml.csproj]: This should also target netcoreapp2.1 to check the portable System.Xaml.

File: src/CoreWf/Expressions/AndAlso.cs (hunk 3 above)
Comment (lbargaoanu) [reply src/CoreWf/Expressions/AndAlso.cs]: Ok, that was the initial code :) I think, just remove the pragma for now. Perhaps we should enable the attributes at some point.

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj (hunk 1 above)
Comment (dmetzgar) [reply src/Test/TestCases.Xaml/TestCases.Xaml.csproj]: Done

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj (hunk 4 above)
Comment (dmetzgar) [reply src/Test/TestCases.Xaml/TestCases.Xaml.csproj]: Done

File: src/CoreWf/Expressions/AndAlso.cs (hunk 3 above)
Comment (dmetzgar) [reply src/CoreWf/Expressions/AndAlso.cs]: Removed

File: src/Test/TestCases.Xaml/TestCases.Xaml.csproj (hunk 2 above)
Comment (dmetzgar) [reply src/Test/TestCases.Xaml/TestCases.Xaml.csproj:1]: Let me handle that in a separate PR. Created issue #76

**Pull request merged**